### PR TITLE
Review pass by Mirja

### PR DESCRIPTION
**Pull request by mirjak**
This review also contains feedback from our ATSSS experts: some edits are made to consider the fact that ATSSSv1 does not only have the MPTCP based solution but also ATSSS-LL (without going into any details of that as it is not relevant here). Further, only the rules to the UE are called ATSSS rules while the rules to the UPF are called PFCP rules. Therefore I removed the "ATSSS" and only talk about "rules" now. We could further clarify this by adding the following text but I don't think this is necessarily needed:

"Note that, as per 3GPP definition, the rules provided to the UE are called ATSSS Rules, while the ATSSS information provided to the UPF is carried via Multi-Access Rules, but for simplicity this document uses the term “ATSSS rules” for the ATSSS information provided to both UE and UPF."

### CODE REVIEW

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 1]
```diff
@@ -364,11 +359,11 @@ The traffic that matches an ATSSS rule can be distributed among available access
...
 
-In order to provide the above-mentioned steering modes, the ATSSS service relies upon a dedicated protocol called Performance Measurement Function (PMF) protocol. This protocol is enabled between the UE and the UPF and is meant to provide RTT measurements and reporting access availability/unavailability by the UE to the UPF. PMF is not specific to a given ATSSS Phase {{sec-phases}}. 
+In order to provide the above-mentioned steering modes, measurement information about the current network state of each path are needed. Often if a multipath capabale protocol is used, the measurements are available as part of the protocol itself. For ATSSS approaches where this is not the case, a dedicated protocol called Performance Measurement Function (PMF) protocol can be uased. This protocol is enabled between the UE and the UPF in order to provide RTT measurements and report access availability/unavailability by the UE to the UPF.
```
Comment (boucadair) [inline draft-bonaventure-quic-atsss-overview.mkd]:
```suggestion
In order to provide the above-mentioned steering modes, measurement information about the current network state of each path are needed. Often if a multipath capable protocol is used, the measurements are available as part of the protocol itself. For ATSSS approaches where this is not the case, a dedicated protocol called Performance Measurement Function (PMF) protocol can be uased. This protocol is enabled between the UE and the UPF in order to provide RTT measurements and report access availability/unavailability by the UE to the UPF. 
```

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 2]
```diff
@@ -364,11 +359,11 @@ The traffic that matches an ATSSS rule can be distributed among available access
...
+In order to provide the above-mentioned steering modes, measurement information about the current network state of each path are needed. Often if a multipath capabale protocol is used, the measurements are available as part of the protocol itself. For ATSSS approaches where this is not the case, a dedicated protocol called Performance Measurement Function (PMF) protocol can be uased. This protocol is enabled between the UE and the UPF in order to provide RTT measurements and report access availability/unavailability by the UE to the UPF. 
 
-Furthermore, these modes of operations were designed assuming that multipath protocols
-such as MPTCP {{RFC6824}} would be used over the different paths between
-the UE and the ATSSS. The implementations of these protocols include two
+These modes of operations can used by multipath protocols
```
Comment (boucadair) [inline draft-bonaventure-quic-atsss-overview.mkd]:
```suggestion
These modes of operations can be met by using multipath protocols
```

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 3]
```diff
@@ -586,15 +581,15 @@ Another approach is to use one QUIC connection over each access network.
...
+based on the rules provided by the network. 
 
 This approach can support steering (by selecting just one connection for each flow), switching 
 (by moving flows from one QUIC connection to the other) as well as splitting when both
 tunnels are used simultaneously for different packets of the same flow. 
 
 However, this approach for splitting is challenging
 since data from the same flow are sent over different connections,
-again using unreliable datagram to minimize head-of-line blocking.
+again using unreliable datagram to minimise head-of-line blocking.
```
Comment (boucadair) [inline draft-bonaventure-quic-atsss-overview.mkd]:
```suggestion
again using unreliable datagram to minimize head-of-line blocking.
```

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 4]
```diff
@@ -647,11 +642,11 @@ TCP when combining different access networks {{MPQUIC-Conext}}.
 
 Steering can be also supported, similarly to MPTCP. The path scheduler can
 map datagrams carrying an entire flow to one or another access network based on the
-ATSSS rules. 
+provided rules. 
 
-Switching can be improved by splitting traffic simultaneously over
+Switching can be improved by splitting traffic simulaneously over
```
Comment (boucadair) [inline draft-bonaventure-quic-atsss-overview.mkd]:
```suggestion
Switching can be improved by splitting traffic simultaneously over
```

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 5]
```diff
@@ -647,11 +642,11 @@ TCP when combining different access networks {{MPQUIC-Conext}}.
...
 both link first such that the congestion window of the new path can be open before
-the old path goes away. This makes handovers smoother, as experience with Multipath TCP on
+the old path goes aways. This makes handovers smoother, as experience with Multipath TCP on
```
Comment (boucadair) [inline draft-bonaventure-quic-atsss-overview.mkd]:
```suggestion
the old path goes away. This makes handovers smoother, as experience with Multipath TCP on
```

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 6]
```diff
@@ -709,19 +704,19 @@ Both proposals rely upon single QUIC connections and inherit thus the same issue
...
 
-In 3GPP architectures, a variety of encapsulations are already used to carry user data. The use of QUIC as a tunneling method for ATSSS purposes will add an extra overhead. This overhead is even exacerbated when the user data is forwarded over a non-3GPP network access given that an IPsec tunnel is used to graft the UE to the 5G Core Network. For example, the overhead without taking into account the QUIC tunnel is at least 142 octets for an IPv6 packet forwarded over a non-3GPP network access. This compares unfavorably to ATSSS Phase 1, which does not add any overhead. 
+In 3GPP architectures, a variety of encapsulations are already used to carry user data. The use of QUIC as a tunneling method for ATSSS will add some additional overhead. When the user data is forwarded over a non-3GPP network access, this overhead comes further in addition to an IPsec tunnel between the UE to the 5G Core Network which is already at least 142 octets for an IPv6 packet forwarded over a non-3GPP network access. In contrast, the MPTCP based mechanism in ATSSS Phase 1, does only add some minor overheading during connection establishment, but addiotional over during the rest of the connection.
```
Comment (boucadair) [inline draft-bonaventure-quic-atsss-overview.mkd]:
```suggestion
In 3GPP architectures, a variety of encapsulations are already used to carry user data. The use of QUIC as a tunneling method for ATSSS will add some additional overhead. When the user data is forwarded over a non-3GPP network access, this overhead comes further in addition to an IPsec tunnel between the UE to the 5G Core Network which is already at least 142 octets for an IPv6 packet forwarded over a non-3GPP network access. In contrast, the MPTCP based mechanism in ATSSS Phase 1 does only add a minor overhead during connection establishment, but no additional overhead during the rest of the connection. 
```

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 7]
```diff
@@ -709,19 +704,19 @@ Both proposals rely upon single QUIC connections and inherit thus the same issue
...
 
 More investigation is required to assess whether the ATSSS Phase 2 overhead is an issue. Solutions to optimize that overhead could be considered, if needed. 
 
 ## Multiple Encryptions
 
-QUIC data that matches an ATSSS rule will be subject to multiple instances of encryptions. Some of these encryptions are redundant (e.g., ATSSS encryption vs. IP Encapsulating Security Payload (ESP)).
+The use of QUIC as a tunneling protocol in ATSSS with add an additional layer of encryption. As there is already encryptions on other layers (e.g.  IP Encapsulating Security Payload (ESP)) this leads to multiple layer of encryption with might be redundant.
```
Comment (boucadair) [inline draft-bonaventure-quic-atsss-overview.mkd]:
```suggestion
The use of QUIC as a tunneling protocol in ATSSS will add an additional layer of encryption. As there is already encryptions on other layers (e.g., IP Encapsulating Security Payload (ESP)) this leads to multiple layer of encryption with might be redundant.
```

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 8]
```diff
@@ -731,47 +726,40 @@ QUIC encapsulated in QUIC or SCREAM encapsulated in QUIC) as well as the coexist
...
 It is assumed that this is a solvable task similar to {{MPDCCP-paper}}, but is
 probably left to the implementer, to take care. Even if the reconstruction of the packet order does not
 become a standardized part of the MP-QUIC in {{sec-mpquic-tunnel}}, it possibly
 requires path sequencing and end-to-end sequencing. 
 
-# QUIC Version 1 Gap Analysis for ATSSS Phase 2 {#gap-analysis}
+# QUICv1 Gap Analysis for ATSSS Phase 2 {#gap-analysis}
 
-These are additional QUIC protocol capabilities that would be useful for ATSSS Phase 2, as described in {{sec-ver2}}, along with our understanding of their maturity level.
+This section summerized QUIC protocol capabilities that would be beneficial for ATSSS Phase 2, as described in {{sec-ver2}}.
```
Comment (boucadair) [inline draft-bonaventure-quic-atsss-overview.mkd]:
```suggestion
This section summarizes QUIC protocol capabilities that would be beneficial for ATSSS Phase 2, as described in {{sec-ver2}}.
```

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 9]
```diff
@@ -731,47 +726,40 @@ QUIC encapsulated in QUIC or SCREAM encapsulated in QUIC) as well as the coexist
...
 
-- ATSSS Phase 2 will need transport services for Ethernet frames and IP packets (with the intention of supporting TCP, UDP, and UDP-encapsulated transport protocols such as QUIC). QUIC Version 1 provides transport services for HTTP/3 but does not yet provide support for tunnelling Ethernet or IP directly over QUIC, as envisioned in ATSSS Phase 2. The QUIC working group is not chartered to provide this capability, although proposals exist.
+- ATSSS Phase 2 is focused on transport services for Ethernet frames and IP packets (with the intention of supporting TCP, UDP, and UDP-encapsulated transport protocols such as QUIC). The discussed approaches are based on tunnelling Ethernet or IP directly over QUIC. The MASQUE working group that is currently in ther charter process is scoped to covery UDP and IP proxying. While UDP proxying does cover the most important use case to support ATSSS for UDP/QUIC, a more generic solution based on IP proxying would simplify the ATSSS design. However, IP proxying is only considered at a later stage in the MASQUE group.
```
Comment (boucadair) [inline draft-bonaventure-quic-atsss-overview.mkd]:
```suggestion
- ATSSS Phase 2 is focused on transport services for Ethernet frames and IP packets (with the intention of supporting TCP, UDP, and UDP-encapsulated transport protocols such as QUIC). The discussed approaches are based on tunnelling Ethernet or IP directly over QUIC. The masque working group that is currently in the chartering validation process is scoped to cover UDP and IP proxying. While UDP proxying does cover the most important use case to support ATSSS for UDP/QUIC, a more generic solution based on IP proxying would simplify the ATSSS design. However, IP proxying is only considered at a later stage by the masque working group. Also, multipath mechanisms of QUIC are not covered in the proposed charter.   
```

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 10]
```diff
@@ -731,47 +726,40 @@ QUIC encapsulated in QUIC or SCREAM encapsulated in QUIC) as well as the coexist
...
 
-- QUIC Version 1 does not provide a multipath transport capability similar to Multipath TCP, as used in ATSSS Phase 1. The QUIC working group is chartered to produce a multipath extension document by December 2021. Proposals exist, although the working group has not yet adopted a draft as the basis for its work. 
+- While the base protocol for QUICv1 does not provide support for unreliable datagrams, an QUIC extension for datagram support has been adaopted by the group and the QUIC working group is chartered to produce this capability by March 2021. This can be usedto support for the additional user-plane protocols as envisioned in ATSSS Phase 2.
```
Comment (boucadair) [inline draft-bonaventure-quic-atsss-overview.mkd]:
```suggestion
- While the base protocol for QUICv1 does not provide support for unreliable datagrams, an QUIC extension for datagram support has been adopted by the group and the QUIC working group is chartered to produce this capability by March 2021. This can be used to support for the additional user-plane protocols as envisioned in ATSSS Phase 2.
```

I like the OLD wording, though.

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 11]
```diff
@@ -788,6 +776,8 @@ Multipath QUIC-specific security considerations can be found in Section 8 of {{I
 
 Section 6 of {I-D.ietf-quic-datagram} discusses security considerations specific to the use of the Unreliable Datagram Extension to QUIC.
 
+This document does not explicitly discuss security consideration specific to the use of QUIC as a tunneling protocol towards a ATSSS proxy, as the intention of this document is to provide an informational overview of the on-going work in 3GPP on ATSSS to support non-TCP, rather than discussing a detailed solution.
```
Comment (boucadair) [inline draft-bonaventure-quic-atsss-overview.mkd]:
```suggestion
This document does not specify specific mechanisms to use QUIC as a tunneling protocol towards a ATSSS proxy, as the intention of this document is to provide an informational overview of the ongoing work in 3GPP on ATSSS to support non-TCP, rather than discussing a detailed solution. Nevertheless, this document cites candidate solutions to provide such tunneling service. Security considerations specific to these solutions are provided below.  
```

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 12]
```diff
@@ -45,11 +45,6 @@ author:
   organization: Tencent America
   email: spencerdawkins.ietf@gmail.com
  -
-  ins: M. Kuehlewind
```
Comment (boucadair) [inline draft-bonaventure-quic-atsss-overview.mkd]: You may want revert this change.

**Review by boucadair — CHANGES_REQUESTED** [review]
I prefer the use or "ATSSS rules" rather than "rules".

**Comment by boucadair** [comment]
> "Note that, as per 3GPP definition, the rules provided to the UE are called ATSSS Rules, while the ATSSS information provided to the UPF is carried via Multi-Access Rules, but for simplicity this document uses the term “ATSSS rules” for the ATSSS information provided to both UE and UPF."

I like this. Thanks.

File: draft-bonaventure-quic-atsss-overview.mkd (hunk 12 above)
Comment (mirjak) [reply draft-bonaventure-quic-atsss-overview.mkd]: Yes, I saw this and did :-)

Comment (mirjak) [reply draft-bonaventure-quic-atsss-overview.mkd]: Actually I was not sure if I should list myself as author as we already have way too many authors. I think it would be more appropriate to list some people as contributor in a separate section at the end.

**Review by boucadair — COMMENTED** [review]
With this note, we may revert to the use of "ATSSS rules" instead of "rules". No?

File: draft-bonaventure-quic-atsss-overview.mkd (hunk 12 above)
Comment (boucadair) [reply draft-bonaventure-quic-atsss-overview.mkd]: We can keep that list, but add a tag for editors as suggested in #81. 

I will make a PR with a proposal.

**Comment by mirjak** [comment]
Yes.. more work for me :-( ... I reverted it in the figure and at only one place in the text as the rest does read nicely now...

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 13]
```diff
@@ -364,11 +364,11 @@ The traffic that matches an ATSSS rule can be distributed among available access
...
 
-In order to provide the above-mentioned steering modes, the ATSSS service relies upon a dedicated protocol called Performance Measurement Function (PMF) protocol. This protocol is enabled between the UE and the UPF and is meant to provide RTT measurements and reporting access availability/unavailability by the UE to the UPF. PMF is not specific to a given ATSSS Phase {{sec-phases}}. 
+In order to provide the above-mentioned steering modes, measurement information about the current network state of each path are needed. Often if a multipath capable protocol is used, the measurements are available as part of the protocol itself. For ATSSS approaches where this is not the case, a dedicated protocol called Performance Measurement Function (PMF) protocol can be uased. This protocol is enabled between the UE and the UPF in order to provide RTT measurements and report access availability/unavailability by the UE to the UPF.
```
Comment (mpiraux) [inline draft-bonaventure-quic-atsss-overview.mkd]:
```suggestion
In order to provide the above-mentioned steering modes, measurement information about the current network state of each path are needed. Often if a multipath capable protocol is used, the measurements are available as part of the protocol itself. For ATSSS approaches where this is not the case, a dedicated protocol called Performance Measurement Function (PMF) protocol can be used. This protocol is enabled between the UE and the UPF in order to provide RTT measurements and report access availability/unavailability by the UE to the UPF. 
```

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 14]
```diff
@@ -731,47 +731,40 @@ QUIC encapsulated in QUIC or SCREAM encapsulated in QUIC) as well as the coexist
...
-without any multipath extension, packet reordering can be assumed to be very
-limited. Therefore most piggybacked services on top of a Unreliable Datagram
-Extension QUIC tunnel which is split as in {{sec-quic-multiple-tunnels}} and
-{{sec-mpquic-tunnel}} are not expecting massive packet reordering and degrades
-their performance if it happens anyway. These potentially are all those services
-that rely on {{RFC0793}} and {{I-D.ietf-quic-transport}} or
-applications that are sensitive to reordering due to real-time demand.
+splitting mode for simultaneously sending data over multiple paths. In this case packet reordering 
+can occur when a QUIC tunnel
+communication is split across paths with very differnet latencies.
```
Comment (mpiraux) [inline draft-bonaventure-quic-atsss-overview.mkd]:
```suggestion
communication is split across paths with very different latencies.
```

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 15]
```diff
@@ -731,47 +731,40 @@ QUIC encapsulated in QUIC or SCREAM encapsulated in QUIC) as well as the coexist
...
+Generally, applications have to have a way deal with packet reordering, since the best
```
Comment (mpiraux) [inline draft-bonaventure-quic-atsss-overview.mkd]:
```suggestion
Generally, applications have to deal with packet reordering, since the best
```

File: draft-bonaventure-quic-atsss-overview.mkd (hunk 7 above)
Comment (boucadair) [reply draft-bonaventure-quic-atsss-overview.mkd]: This nit is to be fixed

File: draft-bonaventure-quic-atsss-overview.mkd (hunk 6 above)
Comment (boucadair) [reply draft-bonaventure-quic-atsss-overview.mkd]: Please fix this one as well. Thanks.

**Comment by SpencerDawkins** [comment]
Just chiming in late (I went back to bed about three hours ago, sorry to miss all the fun), but on the question of adding the "Multi-Access Rules"/"ATSSS Rules" caveat, I think that's definitely helpful. 

The "intended readership" is the IETF community (mostly in QUIC), but this draft is already in at least two of the five contributions in SA2#139E next week, so anything that makes it clear to SA2 that we understand the simplifications we're making without confusing the IETF readers is a good thing.

**Review by mpiraux — COMMENTED** [review]
I had to stop reading before the reordering section unfortunately. I will continue later today.

File: draft-bonaventure-quic-atsss-overview.mkd (hunk 13 above)
Comment (boucadair) [reply draft-bonaventure-quic-atsss-overview.mkd]: s/measurement information about the current network state of each path are needed/measurement information about the current network state of each path is needed

**Comment by markusa** [comment]
Hi Mirja,

I'm just wondering why you propose to rewrite the section "Packet Order Reconstruction for (MP-)QUIC Splitting Mode"? From my view it would become more minimalistic now and leave some information out.

In https://github.com/obonaventure/draft-quic-atsss-reqs/pull/88 was some consesus.

**Comment by mirjak** [comment]
Hey Markus, sorry I didn't review #88 before it was merged. The intention of my rewrite was mainly to clarify, as I had some problem phrasing some of the points correctly. It was not my intention so leave something out. What's the information missing now? Maybe I can re-add it?

**Comment by markusa** [comment]
> 
> 
> Hey Markus, sorry I didn't review #88 before it was merged. The intention of my rewrite was mainly to clarify, as I had some problem phrasing some of the points correctly. It was not my intention so leave something out. What's the information missing now? Maybe I can re-add it?

Most easiest way, you could revert to https://github.com/obonaventure/draft-quic-atsss-reqs/pull/88 and we discuss your potential unclarities.

I miss at least

- comparison to MPTCP in respect to packet loss
- references to {{sec-quic-multiple-tunnels}} and {{sec-mpquic-tunnel}}
- Relation to the unreliable extension

However I like your point:
`This can cause delays in the reordering logic, which in turn can have a negative impact on application that do not require in-order delivery, auch as real-time transmissions.` 
but I would rather say this is also related to piggybacked TCP or QUIC traffic!

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 16]
```diff
@@ -709,19 +709,19 @@ Both proposals rely upon single QUIC connections and inherit thus the same issue
...
 
 More investigation is required to assess whether the ATSSS Phase 2 overhead is an issue. Solutions to optimize that overhead could be considered, if needed. 
 
 ## Multiple Encryptions
 
-QUIC data that matches an ATSSS rule will be subject to multiple instances of encryptions. Some of these encryptions are redundant (e.g., ATSSS encryption vs. IP Encapsulating Security Payload (ESP)).
+The use of QUIC as a tunneling protocol in ATSSS will add an additional layer of encryption. As there is already encryptions on other layers (e.g., IP Encapsulating Security Payload (ESP)) this leads to multiple layer of encryption with might be redundant.
```
Comment (boucadair) [inline draft-bonaventure-quic-atsss-overview.mkd]:
```suggestion
The use of QUIC as a tunneling protocol in ATSSS will add an additional layer of encryption. As there is already encryptions on other layers (e.g., IP Encapsulating Security Payload (ESP)) this leads to multiple layers of encryption, which might be redundant.
```

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 17]
```diff
@@ -731,47 +731,40 @@ QUIC encapsulated in QUIC or SCREAM encapsulated in QUIC) as well as the coexist
...
 
-- We recognize that Multipath QUIC may not provide reassembly or appropriate sequencing for the encapsulated protocols. 
+- When QUIC is used as a tunneling protocol, nested congestion control mechanisms (such as QUIC encapsulated in QUIC) have implications that require further study. 
 
-- We recognize that nested congestion control mechanisms (such as QUIC encapsulated in QUIC) have implications that require further study. 
-
-- We recognize that the complete ATSSS Phase 2 protocol stack would include encrypted headers at multiple levels, and that if this is a problem for ATSSS, it is likely 3GPP's problem to solve. Likewise, the implications of the overhead of the various encapsulations are to be further assessed within 3GPP.
+- When QUIC is used as a tunneling protocol, the complete ATSSS Phase 2 protocol stack would include encrypted headers at multiple layers. It needs further investigation if this is a problem for ATSSS, however, it is likely a problem that can be solve in the 3GPP architecture. Likewise, the implications of the various encapsulation overhead is to be further assessed within 3GPP.
```
Comment (boucadair) [inline draft-bonaventure-quic-atsss-overview.mkd]:
```suggestion
- When QUIC is used as a tunneling protocol, the complete ATSSS Phase 2 protocol stack would include encrypted headers at multiple layers. It needs further investigation if this is a problem for ATSSS, however, it is likely a problem that can be solved by the 3GPP. Likewise, the implications of the various encapsulation overhead is to be further assessed within 3GPP.
```

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 18]
```diff
@@ -788,6 +781,8 @@ Multipath QUIC-specific security considerations can be found in Section 8 of {{I
 
 Section 6 of {I-D.ietf-quic-datagram} discusses security considerations specific to the use of the Unreliable Datagram Extension to QUIC.
 
+This document does not specify specific mechanisms to use QUIC as a tunneling protocol towards a ATSSS proxy, as the intention of this document is to provide an informational overview of the ongoing work in 3GPP on ATSSS to support non-TCP, rather than discussing a detailed solution. Nevertheless, this document cites candidate solutions to provide such tunneling service. Security considerations specific to these solutions are provided below.
```
Comment (boucadair) [inline draft-bonaventure-quic-atsss-overview.mkd]:
```suggestion
This document does not specify specific mechanisms to use QUIC as a tunneling protocol towards an ATSSS proxy, as the intention of this document is to provide an informational overview of the ongoing work in 3GPP on ATSSS to support non-TCP, rather than discussing a detailed solution. Nevertheless, this document cites candidate solutions to provide such tunneling service. Security considerations specific to these solutions are provided below.  
```

**Comment by mirjak** [comment]
@markusa I added now

"While in the case when using the MPTCP converter, losses are retransmitted quickly on the local segment, when the Unreliable Datagram Extension for QUIC is used, the reconstruction mechanism has to further account for packet loss which may occur for any of the paths within the QUIC tunnel."

I'm not sure why the reference are needed. And also not sure what you mean by "Relation to the unreliable extension"; isn't that also covered by this sentence?

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 19]
```diff
@@ -1,5 +1,5 @@
 ---
-title: 3GPP Access Traffic Steering, Switching, and Splitting (ATSSS) Overview for IETF Participants
+title: 3GPPP Access Traffic Steering, Switching, and Splitting (ATSSS) Overview for IETF Participants
```
Comment (obonaventure) [inline draft-bonaventure-quic-atsss-overview.mkd]: Why 3GPPP and not 3GPP ?

File: draft-bonaventure-quic-atsss-overview.mkd (hunk 1 above)
Comment (obonaventure) [inline draft-bonaventure-quic-atsss-overview.mkd]: In order to provide the above-mentioned steering modes, measurement informations about the current network state of each path are needed. Often if a multipath capable protocol is used, the measurements are available as part of the protocol itself. For ATSSS approaches where this is not the case, a dedicated protocol called Performance Measurement Function (PMF) protocol can be used. This protocol is enabled between the UE and the UPF in order to provide RTT measurements and report access availability/unavailability by the UE to the UPF.

File: draft-bonaventure-quic-atsss-overview.mkd (hunk 7 above)
Comment (obonaventure) [inline draft-bonaventure-quic-atsss-overview.mkd]: The use of QUIC as a tunneling protocol in ATSSS will add an additional layer of encryption. As there is already encryptions on other layers (e.g., IP Encapsulating Security Payload (ESP)) this leads to multiple layers of encryption with might be redundant.

Comment (obonaventure) [reply draft-bonaventure-quic-atsss-overview.mkd]: typo

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 20]
```diff
@@ -709,19 +704,19 @@ Both proposals rely upon single QUIC connections and inherit thus the same issue
...
+Many applications that are sending unreliable traffic use various congestion control algorithms to detect congestion and adjust their sending rate based on inferred end-to-end latency and loss characteristics. Examples include Adaptive Video Streaming using e.g. SCREAM {{RFC8298}}, or other media streaming applications that use QUIC as transport layer. When using QUIC version 1 with Unreliable Datagram Extension or Multipath QUIC as ATSSS solution, this leads to nested congestion control, where both inner and outer congestion control coexist. When using Multiple QUICv1 tunnels with Unreliable Datagram Extension or MP-QUIC tunneling, the packet scheduler has additional impact on perceived end-to-end latency and loss due to the potential difference of the individual path characteristics.
```
Comment (obonaventure) [inline draft-bonaventure-quic-atsss-overview.mkd]: Many applications that are sending unreliable traffic use various congestion control algorithms to detect congestion and adjust their sending rate based on inferred end-to-end latency and loss characteristics. Examples include Adaptive Video Streaming using e.g. SCREAM {{RFC8298}}, or other media streaming applications that use QUIC as transport layer. When using QUIC version 1 with Unreliable Datagram Extension or Multipath QUIC as ATSSS solution, this leads to nested congestion control, where both inner and outer congestion controls coexist. When using Multiple QUICv1 tunnels with Unreliable Datagram Extension or MP-QUIC tunneling, the packet scheduler could have additional impact on perceived end-to-end latency and loss due to the potential difference of the individual path characteristics.

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 21]
```diff
@@ -731,47 +726,40 @@ QUIC encapsulated in QUIC or SCREAM encapsulated in QUIC) as well as the coexist
...
-{{sec-mpquic-tunnel}} are not expecting massive packet reordering and degrades
-their performance if it happens anyway. These potentially are all those services
-that rely on {{RFC0793}} and {{I-D.ietf-quic-transport}} or
-applications that are sensitive to reordering due to real-time demand.
+splitting mode for simultaneously sending data over multiple paths. In this case packet reordering 
+can occur when a QUIC tunnel
+communication is split across paths with very differnet latencies.
+Generally, applications have to have a way deal with packet reordering, since the best
+effort for Internet traffic has no guarantee to prevent it. However, in practice, 
+packet reordering in the network s assumed to be very limited. Applications
```
Comment (obonaventure) [inline draft-bonaventure-quic-atsss-overview.mkd]: is assumed

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 22]
```diff
@@ -731,47 +726,40 @@ QUIC encapsulated in QUIC or SCREAM encapsulated in QUIC) as well as the coexist
...
+can occur when a QUIC tunnel
+communication is split across paths with very differnet latencies.
+Generally, applications have to have a way deal with packet reordering, since the best
+effort for Internet traffic has no guarantee to prevent it. However, in practice, 
+packet reordering in the network s assumed to be very limited. Applications 
+that require in-order delivery and e.g. rely on TCP or implement a similar mechanism itself can be sensitive to
+high amounts of reordering and expirience decreased performanced. 
+
+As such it is desirable that the respective
+receiver side of the tunnel termination points has the ability to reconstruct of the original packet ordering.
```
Comment (obonaventure) [inline draft-bonaventure-quic-atsss-overview.mkd]: .. to reconstruct the original packet ordering.

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 23]
```diff
@@ -731,47 +726,40 @@ QUIC encapsulated in QUIC or SCREAM encapsulated in QUIC) as well as the coexist
...
+effort for Internet traffic has no guarantee to prevent it. However, in practice, 
+packet reordering in the network s assumed to be very limited. Applications 
+that require in-order delivery and e.g. rely on TCP or implement a similar mechanism itself can be sensitive to
+high amounts of reordering and expirience decreased performanced. 
+
+As such it is desirable that the respective
+receiver side of the tunnel termination points has the ability to reconstruct of the original packet ordering. 
+When the Unreliable Datagram Extension for QUIC is used, the reconstruction mechanism has to further
+account for packet loss which may occur for any of the paths within the QUIC tunnel. This can cause 
+delays in the reordering logic, which in turn can have a negative impact on application that do not require
```
Comment (obonaventure) [inline draft-bonaventure-quic-atsss-overview.mkd]: on applicationS that

File: draft-bonaventure-quic-atsss-overview.mkd [hunk 24]
```diff
@@ -731,47 +726,40 @@ QUIC encapsulated in QUIC or SCREAM encapsulated in QUIC) as well as the coexist
...
+- We envision the ability to select paths (steering), detect path failures and reroute traffic (switching), and forward packets on multiple active paths simultaneously (splitting), based on external policies, including active-standby, smallest delay, weighted load-balancing, and path selection based on assigned priorities, for the full range of encapsulated protocols in ATSSS Phase 2, similar to the abilities provided by Multipath TCP for ATSSS Phase 1. Splitting cannot be supported easily in the discussed QUICv1-based approaches. Multipath transport capability similar to Multipath TCP, as used in ATSSS Phase 1, would support splitting well. The QUIC working group is originally chartered to produce a multipath extension document by December 2021. Proposals exist, however, this work has been postponed to QUICv2 and discussion is still on-going if support will be kept in the chather.
```
Comment (obonaventure) [inline draft-bonaventure-quic-atsss-overview.mkd]: chather -> charter

**Review by obonaventure — CHANGES_REQUESTED** [review]
minor typos in the text, besides this fine for me

**Comment by markusa** [comment]
> 
> 
> @markusa I added now
> 
> "While in the case when using the MPTCP converter, losses are retransmitted quickly on the local segment, when the Unreliable Datagram Extension for QUIC is used, the reconstruction mechanism has to further account for packet loss which may occur for any of the paths within the QUIC tunnel."
> 
> I'm not sure why the reference are needed. And also not sure what you mean by "Relation to the unreliable extension"; isn't that also covered by this sentence?

@mirjak 
Please have a look into the notes from last call. There we agreed that re-construction is completely different from the MPTCP of ATSSS v1 and we should start from this point to highlight the issue.

There are several points now in your rewritten text I disagree.

CC in CC and the reconstruction of the packet order I consider as the most critical points when we talk about implementation. All the other things were proven with MPTCP. In that respect we should handle this topic carefully.

I kindly as you to revert to #88 and we open a separate issue on this? Then it would be much more simpler to move forward with your PR, which I support except from this.

**Review by obonaventure — APPROVED** [review]

**Comment by obonaventure** [comment]
I merge this pull request to have a clean document, Markus' comment has been moved to a separate issue.
